File: C/练习/code_10_1.c
#include <stdio.h>


void print_smFace_with_stair(int *m, int *n) {
    char sm_face = '*';
    char stair = 'n';
    for (int i = 1; i <= *m; i++) {
        printf("%c", sm_face);
        if (i == *m) {
            printf("%c\n", sm_face);
        }
    }
    for (int j = 1; j <= *n; j++) {
        for (int k = 0; k <= j; k++) {
            printf("%c", stair);
            if (k == j) {
                printf("%c\n", stair);
            }
        }
    }
}



int main(void) {
    int m, n;
    printf("请输入笑脸个数和楼梯高度：");
    scanf("%d %d", &m, &n);
    print_smFace_with_stair(&m, &n);
    return 0;
}

File: C/Replicate_With_Python/code_10_1.py
def print_smFace_with_stair(m, n):
    sm_face = chr(9786)
    stair = chr(9656)
    print(sm_face * m)
    for i in range(3, n + 1):
        print(stair * i)
    return 


if __name__ == '__main__':
    m = int(input("请输入笑脸的个数："))
    n = int(input("请输入楼梯的高度："))
    print_smFace_with_stair(m, n)